tests/plugins/test_ddp_plugin.py

```diff
@@ -94,3 +95,39 @@ def creates_children(self):
         RuntimeError, match="Lightning attempted to launch new distributed processes with `local_rank > 0`."
     ):
         trainer.fit(model)
+
+
+@RunIf(min_gpus=1)
+def test_ddp_module_wrapper():
+    """Tests with ddp plugin."""
+    model = BoringModel()
+    ddp_plugin = DDPPlugin()
+    trainer = Trainer(
+        max_epochs=1,
+        gpus=1,
+        plugins=[ddp_plugin],
+    )
+    # test wrap the model if fitting
+    trainer.state.fn = TrainerFn.FITTING
+    trainer.accelerator.connect(model)
+    trainer.accelerator.setup_environment()
+    trainer.accelerator.setup(trainer)
+    trainer.lightning_module.trainer = trainer
```

Comment:
FYI @daniellepintz we'll get much better test isolation by removing the trainer reference from the lightning module. Ideally this is some TrainerState object which can be easily mocked as data